src/popper/runner_host.py

```diff
@@ -293,6 +293,212 @@ def _update_with_engine_config(self, container_args):
                 container_args[k] = update_with[k]
 
 
+class PodmanRunner(StepRunner):
+    """Runs steps in podman on the local machine."""
+
+    def __init__(self, init_podman_client=True, **kw):
+        super(PodmanRunner, self).__init__(**kw)
+
+        self._spawned_containers = set()
+        _, _, self._p_info = HostRunner._exec_cmd(["podman", "info"], logging=False)
+
+        if not init_podman_client:
+            return
+
+        try:
+            self._p_version = HostRunner._exec_cmd(["podman", "version"], logging=False)
+        except Exception as e:
+            log.debug(f"Podman error: {e}")
+            log.fail(f"Unable to connect to podman, is it installed?")
+
+        log.debug(f"Podman info: {pu.prettystr(self._p_info)}")
+
+    def run(self, step):
+        """Executes the given step in podman."""
+        cid = pu.sanitized_name(step.id, self._config.wid)
+        container = self._find_container(cid)
+
+        if not container and self._config.reuse:
+            log.fail(
+                f"Cannot find an existing container for step '{step.id}' to be reused"
+            )
+
+        if container and not self._config.reuse and not self._config.dry_run:
+            cmd = ["podman", "rm", "-f", container]
+            HostRunner._exec_cmd(cmd, logging=False)
+            container = None
+
+        if not container and not self._config.reuse:
+            container = self._create_container(cid, step)
+
+        log.info(f"[{step.id}] podman start")
+
+        if self._config.dry_run:
+            return 0
+
+        self._spawned_containers.add(container)
+
+        try:
+            cmd = ["podman", "start", container]
+            HostRunner._exec_cmd(cmd, logging=False)
```

Comment:
Would we need to store the output in a variable?

Reply:
when `logging=True`, the output is not returned. But we do want to catch the exit code of the command, so something like:

```python
_, ecode, _ = HostRunner._exec_cmd(cmd)
```

so that we can then return the exit code of the container

Reply:
I just saw this. Thanks